api/src/main/java/org/openmrs/api/ObsService.java

```diff
@@ -537,4 +539,51 @@ public Integer getObservationCount(List<Person> whom, List<Encounter> encounters
 			Integer obsGroupId, Date fromDate, Date toDate, boolean includeVoidedObs, String accessionNumber)
 			throws APIException;
 	
+
+	/**
+	 * Creates or updates the given {@link ObsReferenceRange} in the database
+	 *
+	 * @param obsReferenceRange the obsReferenceRange to save
+	 *                             
+	 * @since 2.7.0
+	 * 
+	 * <strong>Should</strong> create a new obs reference range
+	 * 	 * <strong>Should</strong> edit an existing obs reference range
+	 */
+	ObsReferenceRange saveObsReferenceRange(final ObsReferenceRange obsReferenceRange);
+
+	/**
+	 * @param id ObsReferenceRange id
+	 * @return {@link ObsReferenceRange}
+	 *
+	 * @since 2.7.0
+	 *
+	 * <strong>Should</strong> get the ObsReferenceRange with the given id
+	 * <strong>Should</strong> return null if ObsReferenceRange is not found
+	 */
+	@Authorized(PrivilegeConstants.GET_OBS)
+	ObsReferenceRange getObsReferenceRangeById(final Integer id);
```

Comment:
What is the reason behind declaring these parameters as final. Are we doing it else where in our code base?

Reply:
It's just a good practice to make sure that the arg won't change in the method. But it should be in the implementation instead. I have already made this change